src/lib/bootloader/finish_client.rb

```diff
@@ -65,21 +65,25 @@ def write
       # and remember result of merge as current one
       ::Bootloader::BootloaderFactory.current = system
 
-      # fate #303395: Use kexec to avoid booting between first and second stage
-      # copy vmlinuz, initrd and flush kernel option into /var/lib/YaST2
-      if Yast::Linuxrc.InstallInf("kexec_reboot") == "1"
-        kexec = ::Bootloader::Kexec.new
-        kexec.prepare_environment
-      else
-        log.info "Installation started with kexec_reboot set 0"
-      end
+      if !BootloaderFactory.current.is_a?(SystemdBoot)
+        # not for SystemdBoot bootloader
```

Comment:
can you also add reason why kexec and dracut regeneration is not for systemdboot ?

Reply:
A good point. I have added it.